This is a test for the add_integer function in 0-add_integer.py

from 0-add_integer import add_integer

# One input addition
    >>> add_integer(0)
    98
    >>> add_integer(-10)
    88
    >>> add_integer(1.5)
    99
    >>> add_integer(10)
    109

# Two input addition
    >>> add_integer(1, 2)
    3
    >>> add_integer(2, -1)
    1
    >>> add_integer(2.5, 1.5)
    3

# One input addition TypeError
passing a list
    >>>add_integer([])
    Traceback (most recent call last):
    ...
    TypeError: a must be an integer
passing a complex number
    >>> add_integer(12+3j)
    Traceback (most recent call last):
    ...
    TypeError: a must be an integer
passing a string
    >>> add_integer("Efe")
    Traceback (most recent call last):
    ...
    TypeError: a must be an integer
passing a character
    >>> add_integer('i')
    Traceback (most recent call last):
    ...
    TypeError: a must be an integer
passing no arguments
    >>> add_integer(None)
    Traceback (most recent call last):
    ...
    TypeError: a must be an integer

# Two input addition TypeError
passing a number and a string
    >>> add_integer(2, "Efe")
    Traceback (most recent call last):
    ...
    TypeError: b must be an integer
passing a character and a number
    >>> add_integer('c', 3)
    Traceback (most recent call last):
    ...
    TypeError: a must be an integer
pasing a number and a list
    >>> add_integer(1, [])
    Traceback (most recent call last):
    ...
    TypeError: b must be an integer
passing a tuple
    >>> add_integer((1, 2))
    Traceback (most recent call last):
    ...
    TypeError: a must be an integer

Case Overflow:

    >>> add_integer(float('inf'), 0)
    Traceback (most recent call last):
    	      ...
    OverflowError: cannot convert float infinity to integer

Case Overflow 2:

    >>> add_integer(float('inf'), float('-inf'))
    Traceback (most recent call last):
    	      ...
    OverflowError: cannot convert float infinity to integer

Case NaN 1:

    >>> add_integer(0, float('nan'))
    Traceback (most recent call last):
    	      ...
    ValueError: cannot convert float NaN to integer

Case NaN 2:

    >>> add_integer(float('nan'))
    Traceback (most recent call last):
    	      ...
    ValueError: cannot convert float NaN to integer